tests/actions/buildiso_test.py

```diff
@@ -311,3 +344,57 @@ def test_generate_profile(
             result
             == " append initrd=testdistro.img install=http://192.168.1.1:80/cblr/links/testdistro autoyast=default.ks"
         )
+
+        # Arrange
+        folder = create_kernel_initrd(fk_kernel, fk_initrd)
+        kernel_path = os.path.join(folder, fk_kernel)
+        initrd_path = os.path.join(folder, fk_initrd)
+        test_distro = Distro(api)
+        test_distro.name = "testdistro"
+        test_distro.kernel = kernel_path
+        test_distro.initrd = initrd_path
+        api.add_distro(test_distro)
+        test_profile = Profile(api)
+        test_profile.name = "testprofile"
+        test_profile.distro = test_distro.name
+        api.add_profile(test_profile)
+        blendered_data = utils.blender(api, False, test_profile)
+        test_builder = AppendLineBuilder(test_distro.name, blendered_data)
+
+        # Act
+        result = test_builder.generate_profile("redhat", "rhel7")
+
+        # Assert
+        # Very basic test yes but this is the expected result atm
+        # TODO: Make tests more sophisticated
+        assert (
+            result
+            == " append initrd=testdistro.img inst.ks=default.ks"
+        )
+
+        # Arrange
+        folder = create_kernel_initrd(fk_kernel, fk_initrd)
+        kernel_path = os.path.join(folder, fk_kernel)
```

Comment:
I am confused by this bit? Is there a function header missing or is this done on purpose?

Reply:
This is done on purpose, but from your comment, I understand this is not the way it should be done, or at least not the way you expected it to be done.

Reply:
My idea is to have small independent tests. Each test has one arrange, one act and one assert stage. Cleanup should be done in pytest fixtures. Arrange may be a fixture but doesn't have to be. Please try to reuse existing fixtures to reduce the amount of testcode.

Reply:
@xavierba Any chance we can get my wish in this PR? I would really love to have it.